owid/catalog/tables.py

```diff
@@ -408,10 +447,20 @@ def prune_metadata(self) -> "Table":
     def copy(self, deep: bool = True) -> "Table":
         """Copy table together with all its metadata."""
         tab = super().copy(deep=deep)
-        tab.metadata = copy.deepcopy(self.metadata)
-        tab._fields = copy.deepcopy(self._fields)
+        tab.copy_metadata_from(self)
         return tab
 
+    def copy_metadata_from(self, table: "Table") -> None:
+        """Copy metadata from a different table to self."""
+        self.metadata = dataclasses.replace(table.metadata)
+
+        # NOTE: copying with `dataclasses.replace` is much faster than `copy.deepcopy`
```

Comment:
I suppose you have checked that table.metadata is not affected by this.

Reply:
I have some unit tests.